### PR TITLE
feat: Invoice logic refactoring for multiple plans

### PR DESCRIPTION
**Pull request by rsempe**
Base PR: https://github.com/getlago/lago-api/pull/354.

### CODE REVIEW

File: app/services/invoices/create_service.rb [hunk 1]
```diff
@@ -2,36 +2,41 @@
...
       ActiveRecord::Base.transaction do
-        invoice = Invoice.find_or_create_by!(
-          subscription: subscription,
-          from_date: from_date,
-          to_date: to_date,
-          charges_from_date: charges_from_date,
-          issuing_date: issuing_date,
+        invoice = Invoice.create!(
+          customer: customer,
+          issuing_date: (Time.zone.at(timestamp) - 1.day).to_date,
```
Comment (vincent-pochet) [inline app/services/invoices/create_service.rb:18]: What is the reason behind this `- 1.day`.
I have the feeling that could cause issues with invoices for terminated subscription.

**Review by vincent-pochet — COMMENTED** [review]
Looks good globally :rocket: 

I just have two points I would like to raise:
- How this will behave with `current usage` feature
- Not directly linked to this PR, but with the feature in general: We might have an issue with event filtering for upgraded subscription. Let me know if you want to talk about it.

File: app/services/invoices/customer_usage_service.rb [hunk 2]
```diff
@@ -35,10 +35,7 @@ def usage
     def compute_usage
       Rails.cache.fetch(cache_key, expires_in: cache_expiration.days) do
         @invoice = Invoice.new(
-          subscription: subscription,
-          charges_from_date: charges_from_date,
-          from_date: from_date,
-          to_date: to_date,
+          customer: subscription.customer,
```
[the remark this answers is not in the document]
Comment (rsempe) [reply app/services/invoices/customer_usage_service.rb:38]: Here is an example of how it's displayed now:
Capture d’écran 2022-08-04 à 11 08 16

File: app/services/invoices/create_service.rb (hunk 1 above)
Comment (rsempe) [reply app/services/invoices/create_service.rb:18]: I have the feeling that nothing has changed here, previously we were calling `issuing_date = to_date` and `to_date = (Time.zone.at(timestamp) - 1.day).to_date`, no?

**Review by vincent-pochet — APPROVED** [review]

**Comment by vincent-pochet** [comment]
@rsempe Good for me, except it appears specs are broken

**Pull request merged**